crates/api/src/gen.rs

```diff
@@ -0,0 +1,9363 @@
+        Reverted,
+    }
+
+    #[derive(Clone, Debug, Deserialize, Serialize)]
+    pub enum TxnFinalityStatus {
+        #[serde(rename = "ACCEPTED_ON_L2")]
+        AcceptedOnL2,
+        #[serde(rename = "ACCEPTED_ON_L1")]
+        AcceptedOnL1,
+    }
+
+    #[derive(Clone, Debug, Deserialize, Serialize)]
+    pub struct TxnHash(pub Felt);
+
+    #[derive(Clone, Debug, Deserialize, Serialize)]
+    #[serde(untagged)]
+    pub enum TxnReceipt {
+        InvokeTxnReceipt(InvokeTxnReceipt),
+        L1HandlerTxnReceipt(L1HandlerTxnReceipt),
+        DeclareTxnReceipt(DeclareTxnReceipt),
+        DeployTxnReceipt(DeployTxnReceipt),
+        DeployAccountTxnReceipt(DeployAccountTxnReceipt),
+    }
+
+    #[derive(Clone, Debug, Deserialize, Serialize)]
+    pub enum TxnStatus {
+        #[serde(rename = "RECEIVED")]
+        Received,
+        #[serde(rename = "REJECTED")]
+        Rejected,
+        #[serde(rename = "ACCEPTED_ON_L2")]
+        AcceptedOnL2,
+        #[serde(rename = "ACCEPTED_ON_L1")]
+        AcceptedOnL1,
+    }
+
+    #[derive(Clone, Debug, Deserialize, Serialize)]
+    pub enum TxnType {
+        #[serde(rename = "DECLARE")]
+        Declare,
+        #[serde(rename = "DEPLOY")]
+        Deploy,
+        #[serde(rename = "DEPLOY_ACCOUNT")]
+        DeployAccount,
+        #[serde(rename = "INVOKE")]
+        Invoke,
+        #[serde(rename = "L1_HANDLER")]
+        L1Handler,
+    }
+
+    #[derive(Clone, Debug, Deserialize, Serialize)]
+    pub struct TypedParameter {
+        pub name: String,
+        pub r#type: String,
+    }
+
+    #[derive(Clone, Debug, Deserialize, Serialize)]
+    #[serde(try_from = "String")]
+    pub struct U128(String);
+
+    mod u128 {
+        use super::jsonrpc;
+        use super::U128;
+        use once_cell::sync::Lazy;
+        use regex::Regex;
+
+        static U128_REGEX: Lazy<Regex> = Lazy::new(|| {
+            Regex::new("^0x(0|[a-fA-F1-9]{1}[a-fA-F0-9]{0,31})$").unwrap()
+        });
+
+        impl U128 {
+            pub fn try_new(value: &str) -> Result<Self, jsonrpc::Error> {
+                if U128_REGEX.is_match(value) {
+                    Ok(Self(value.to_string()))
+                } else {
+                    Err(jsonrpc::Error {
+                        code: 1001,
+                        message: format!(
+                            "U128 value does not match regex: {value}"
+                        ),
+                    })
+                }
+            }
+        }
+
+        impl TryFrom<String> for U128 {
+            type Error = String;
+            fn try_from(value: String) -> Result<Self, Self::Error> {
+                Self::try_new(&value).map_err(|e| e.message)
+            }
+        }
+
+        impl AsRef<String> for U128 {
+            fn as_ref(&self) -> &String {
+                &self.0
+            }
+        }
+    }
+
+    #[derive(Clone, Debug, Deserialize, Serialize)]
+    #[serde(try_from = "String")]
+    pub struct U64(String);
```

Comment:
Why such newtypes?

Reply:
All the generated code comes directly from the spec. The spec defines such type, so the generated code has it.